<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Run optimized single pathway ASSIGN — runassignGFRN • ASSIGN</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Run optimized single pathway ASSIGN — runassignGFRN" />

<meta property="og:description" content="This function runs eight ASSIGN runs based on the pathway optimizations
from the paper. You can run all eight pathways in serial, or call this
function and specify the run parameter to run a specific pathway.
Some ASSIGN parameters can be customized using this function. The default
values were used in the analysis for the paper." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ASSIGN</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.21.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/ASSIGN.vignette.html">Primer</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/compbiomed/ASSIGN">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Run optimized single pathway ASSIGN</h1>
    
    <div class="hidden name"><code>runassignGFRN.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>This function runs eight ASSIGN runs based on the pathway optimizations
from the paper. You can run all eight pathways in serial, or call this
function and specify the run parameter to run a specific pathway.
Some ASSIGN parameters can be customized using this function. The default
values were used in the analysis for the paper.</p>
    
    </div>

    <pre class="usage"><span class='fu'>runassignGFRN</span>(<span class='no'>indata</span>, <span class='kw'>run</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"akt"</span>, <span class='st'>"bad"</span>, <span class='st'>"egfr"</span>, <span class='st'>"her2"</span>, <span class='st'>"igf1r"</span>,
  <span class='st'>"krasgv"</span>, <span class='st'>"raf"</span>), <span class='kw'>optimized_geneList</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>use_seed</span> <span class='kw'>=</span> <span class='fl'>1234</span>,
  <span class='kw'>sigma_sZero</span> <span class='kw'>=</span> <span class='fl'>0.05</span>, <span class='kw'>sigma_sNonZero</span> <span class='kw'>=</span> <span class='fl'>0.5</span>, <span class='kw'>S_zeroPrior</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>iter</span> <span class='kw'>=</span> <span class='fl'>1e+05</span>, <span class='kw'>burn_in</span> <span class='kw'>=</span> <span class='fl'>50000</span>, <span class='kw'>exclude_common_genes</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>adaptive_S</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>ECM</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>indata</th>
      <td><p>The list of data frames from ComBat.step2</p></td>
    </tr>
    <tr>
      <th>run</th>
      <td><p>specifies the pathways to predict. The default list will
cause all eight pathways to be run in serial. Specify a pathway ("akt",
"bad", "egfr", etc.) or list of pathways to run those pathways only.</p></td>
    </tr>
    <tr>
      <th>optimized_geneList</th>
      <td><p>a list of custom optimized gene lists for the gfrn
pathways either created manually or output by optimizeGFRN</p></td>
    </tr>
    <tr>
      <th>use_seed</th>
      <td><p>Set the seed before running ASSIGN. This will make the result
consistent between runs. The default is 1234. Set use_seed as FALSE to not
set a seed.</p></td>
    </tr>
    <tr>
      <th>sigma_sZero</th>
      <td><p>Each element of the signature matrix (S) is modeled by a
spike-and-slab mixture distribution. Sigma_sZero is the variance of the
spike normal distribution. The default is 0.05.</p></td>
    </tr>
    <tr>
      <th>sigma_sNonZero</th>
      <td><p>Each element of the signature matrix (S) is modeled by
a spike-and-slab mixture distribution. Sigma_sNonZero is the variance of the
slab normal distribution. The default is 0.5.</p></td>
    </tr>
    <tr>
      <th>S_zeroPrior</th>
      <td><p>Logicals. If TRUE, the prior distribution of signature
follows a normal distribution with mean zero. The default is FALSE.</p></td>
    </tr>
    <tr>
      <th>iter</th>
      <td><p>The number of iterations in the MCMC. The default is 100000.</p></td>
    </tr>
    <tr>
      <th>burn_in</th>
      <td><p>The number of burn-in iterations. These iterations are
discarded when computing the posterior means of the model parameters. The
default is 50000.</p></td>
    </tr>
    <tr>
      <th>exclude_common_genes</th>
      <td><p>Remove commonly differentially expressed genes
for overexpression signatures. The default is FALSE.</p></td>
    </tr>
    <tr>
      <th>adaptive_S</th>
      <td><p>Logical. If TRUE, the model adapts the signatures (S) of
genomic measures for the test samples. The default for GFRN analysis is TRUE.</p></td>
    </tr>
    <tr>
      <th>ECM</th>
      <td><p>Logicals. If TRUE, ECM algorithm, rather than Gibbs sampling, is
applied to approximate the model parameters. The default is FALSE.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Data is output to the current working directory in a results
directory.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><span class='co'># NOT RUN {</span>
<span class='no'>testData</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/read.table'>read.table</a></span>(<span class='st'>"https://drive.google.com/uc?authuser=0&amp;id=1mJICN4z_aCeh4JuPzNfm8GR_lkJOhWFr&amp;export=download"</span>,
                       <span class='kw'>sep</span><span class='kw'>=</span><span class='st'>'\t'</span>, <span class='kw'>row.names</span><span class='kw'>=</span><span class='fl'>1</span>, <span class='kw'>header</span><span class='kw'>=</span><span class='fl'>1</span>)
<span class='no'>combat.data</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='ComBat.step2.html'>ComBat.step2</a></span>(<span class='no'>testData</span>, <span class='kw'>pcaPlots</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)
<span class='fu'>runassignGFRN</span>(<span class='no'>combat.data</span>)
<span class='co'># }</span><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Ying Shen, Andrea H. Bild, W. Evan Johnson, Mumtehena Rahman, David Jenkins.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

